File: pc-square-io-separator-java/src/main/java/com/mj/pc/square/io/separator/java/constant/ResourcePath.java
package com.mj.pc.square.io.separator.java.constant;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

/**
 * @author Montaser Sbaih
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 15-10-2022
 */

@NoArgsConstructor(access = AccessLevel.PRIVATE)
final class ResourcePath {

    //==================================================={Interfaces}===================================================
    public static final String INTERFACE = "/interface/";

    public static final String DIALOG = INTERFACE + "dialog/";

    public static final String LAYOUT = INTERFACE + "layout/";

    //====================================================={Assist}=====================================================
    private static final String ASSIST = "/assist/";

    public static final String STYLE_SHEET = ASSIST + "css/";

    public static final String STYLE_CONTROL = STYLE_SHEET + "control/";

    public static final String STYLE_USER_INTERFACE = STYLE_SHEET + "interface/";

    public static final String STYLE_DIALOG_INTERFACE = STYLE_USER_INTERFACE + "dialog/";

    public static final String STYLE_LAYOUT_INTERFACE = STYLE_USER_INTERFACE + "layout/";

    public static final String ICON = ASSIST + "icon/";

    public static final String IMAGE = ASSIST + "image/";

    //====================================================={Static}=====================================================

    public static final String STATIC = "/static/";
}

File: pc-square-io-separator-kotlin/src/main/kotlin/com/mj/pc/square/io/separator/kotlin/constant/ResourcePath.kt
package com.mj.pc.square.io.separator.kotlin.constant

import lombok.AccessLevel
import lombok.NoArgsConstructor

/**
 * @author Montaser Sbaih
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 15-10-2022
 */

@NoArgsConstructor(access = AccessLevel.PRIVATE)
internal object ResourcePath {

    //==================================================={Interfaces}===================================================
    const val INTERFACE = "/interface/"

    const val DIALOG = INTERFACE + "dialog/"

    const val LAYOUT = INTERFACE + "layout/"

    //====================================================={Assist}=====================================================
    private const val ASSIST = "/assist/"

    const val STYLE_SHEET = ASSIST + "css/"

    const val STYLE_CONTROL = STYLE_SHEET + "control/"

    const val STYLE_USER_INTERFACE = STYLE_SHEET + "interface/"

    const val STYLE_DIALOG_INTERFACE = STYLE_USER_INTERFACE + "dialog/"

    const val STYLE_LAYOUT_INTERFACE = STYLE_USER_INTERFACE + "layout/"

    const val ICON = ASSIST + "icon/"

    const val IMAGE = ASSIST + "image/"

    //====================================================={Static}=====================================================
    const val STATIC = "/static/"
}